### DISCUSSION TITLE
Advice on migrating from Windows to Linux

### DISCUSSION THREAD

**Discussion by salaivv** [abraunegg/onedrive#1263] [Q&A] [answered]
Hi @abraunegg,

I have been trying out your OneDrive client for Linux on a virtual machine. So far, it is working great. Also, I like the ability to manually synchronize when needed. I usually pause OneDrive sync for the entire day and only sync at the end so that Windows doesn't continuously throttle up my fans. :)

Thank you so much for your efforts in building this! Looking forward to more.

So now, feeling confident that I can continue using OneDrive, I am planning to migrate entirely to Linux. But I have a few questions. 

I currently have my OneDrive folder in my data partition (D:\OneDrive). Now, if I overwrite my Windows partition with a Linux install and configure your client to use the same OneDrive folder in my data partition, will it retain all the files in that folder intact or shall I have to freshly fetch all the files from the cloud back to my PC for the sync to work properly?

And, in case it can retain the files in place, will it create a state cache for all the files in the OneDrive folder when I run sync for the first time?

**Comment by abraunegg** [comment, marked as answer]
@salaivv 
> I currently have my OneDrive folder in my data partition (D:\OneDrive). Now, if I overwrite my Windows partition with a Linux install and configure your client to use the same OneDrive folder in my data partition, will it retain all the files in that folder intact or shall I have to freshly fetch all the files from the cloud back to my PC for the sync to work properly?

This depends. Your D:\ will most likely be formatted as NTFS. Reading and writing NTFS on Linux is possible, however there will be quirks and things that you may uncover that might cause you problems. You are potentially better off backing up your *entire* system and wiping everything and starting clean. Make 2 backups to be sure. NTFS also does not support 'inotify' which is also needed to detect local file system changes real-time when running with `--monitor` or systemd services.

When configuring the client on Linux, edit your `config` file and edit the 'sync_dir' parameter to where your OneDrive data now sits:
```
sync_dir = "/path/to/my/data/"
```

The client on first startup will sync, but check & determine that the file is already present locally, populate the cache file and you are on your way.

Always *test* first with `--synchronize --dry-run` before activating any systemd service file for constant background syncing.

Additionally if you *really* care about your data integrity on your Linux system, use [ZFS](https://zfsonlinux.org/) as your filesystem. If your chosen Linux distribution does not support / include it / have options for you to add it, I strongly advise that you find one that does. You want to ensure your data on your system has as much protection as possible from data loss events such as silent data corruption / physical disk errors and more, and using ZFS is a sure fire way to achieve this.

Further to this, implement a *good* backup schedule / process for your data - backup to an external drive + off-site (and no - backing up to OneDrive is not what I am talking about here) - backup your file system(s), archive, store as an encrypted file on AWS S3 | Backblaze | Google Cloud Storage - so that in the event of something *really bad happening* (like fire, theft, total system failure from power surge) you can get all your data / system / configuration back from these files.

**Reply by salaivv** [reply]
Thank you @abraunegg for your detailed explanation.

I never knew file systems would limit certain functionality! As you said I will wipe my drive after backing up my files and do a clean install. Also will look into setting up ZFS – sounds interesting.

Once again, thank you so much for helping out!
